runtime/src/qio/qio_popen.c

```diff
@@ -576,3 +578,78 @@ qioerr qio_proc_communicate(
   return err;
 }
 
+/* Send a signal to the specified pid.
+
+   The signal value `qio_sig` is not the actual signal value defined in
+   signal.h.  The qio_sig value is Chapel-specific and must be converted to
+   the system-specific signal value.
+ */
+qioerr qio_send_signal(int64_t pid, int qio_sig)
+{
+  qioerr err = 0;
+  int rc = 0;
+  int sig = 0;
+
+  // Convert qio_sig values (from Chapel signal enums) to system-specific
+  // signal values.  This isn't ideal, but we don't have extern params.
```

Comment:
Yeah, this is a big block of code that it would be better not to need to write.
Is there a reason you couldn't use extern const for all of these signals?
For example the way that SysBasic.chpl manages system error codes?
If it's so that you get things scoped in an enum - I'm not really worried about
things like SIGABRT being exported from the Spawn module. I can see
benefits to using an enum but I think I'd prefer an approach that more simply
corresponds to the C situation - where e.g. SIGABRT is a top-level constant.
The module code would be e.g.:

      extern const SIGINFO:c_int;

and then this code would just take in `sig` instead of `qio_sig`.
For the platform-specific signals, my advice would be to make qio_send_signal
take in the system signal number (e.g. SIGINFO) but not to create a Chapel
variable/enum entry for SIGINFO. People wanting to write platform-specific
Chapel code with such a signal can figure out the number themselves e.g.
with

      extern const SIGINFO:c_int